apps/src/gamelab/AnimationPicker/actions.js

```diff
@@ -12,9 +12,11 @@ var AnimationPickerAction = module.exports.AnimationPickerAction = {
  * Show the animation picker
  * @returns {{type: AnimationPickerAction}}
  */
-module.exports.showAnimationPicker = function () {
+module.exports.showAnimationPicker = function (onComplete, onCancel) {
   return {
-    type: AnimationPickerAction.SHOW_ANIMATION_PICKER
+    type: AnimationPickerAction.SHOW_ANIMATION_PICKER,
+    onComplete: onComplete,
+    onCancel: onCancel
```

Comment:
I find myself unsatisfied with Dan Abramov's answer on this one:

http://stackoverflow.com/questions/35623656/how-can-i-display-a-modal-dialog-in-redux-that-performs-asynchronous-actions/35641680

Storing an enum value in the state for which dialog you have open and what it did/is doing is... fine, I guess.  But it seems a lot cleaner to me to just [specify the resulting action as a callback](https://github.com/code-dot-org/code-dot-org/blob/animation-upload/apps/src/gamelab/AnimationTab/actions.js#L15), rather than to write it into a switch statement somewhere.